internal/core/ports/operation_lease_storage.go

```diff
@@ -36,8 +36,10 @@ type OperationLeaseStorage interface {
 	RevokeLease(ctx context.Context, schedulerName, operationID string) error
 	// RenewLease renews an active lease adding the specified TTL to it. NOTE: The lease must exist (by calling GrantLease) before renewing it.
 	RenewLease(ctx context.Context, schedulerName, operationID string, ttl time.Duration) error
-	// FetchLeaseTTL fetches the time the lease will expire given the operation.
+	// FetchLeaseTTL fetches the time the lease will expire given the operation. Return error if the lease does not exist.
```

Comment:
@lyracampos @caiordjesus What do you guys think about letting the FetchLeaseTTL method here? should we delete it or not?

Reply:
I still se advantage in having a method like this here, with new `FetchOperationsLease` method we reduce the number of redis query to 1, but the time complexity of the method it uses is linear [O(n)](https://redis.io/commands/zmscore) while the method FetchLeaseTTL is constant [O(1)](https://redis.io/commands/zscore). In a scenario where we want to reduce the overall query time (given a lease list) and not the number of connections, we could use FetchLeaseTTL method launching a goroutine for each operation id. The average time complexity would be O(1) vs the O(n) of FetchOperationsLease.

Reply:
100% agree!